src/internal/connector/onedrive/drive.go

```diff
@@ -172,22 +183,41 @@ func collectItems(
 	pager itemPager,
 	driveID, driveName string,
 	collector itemCollector,
-) (string, map[string]string, map[string]struct{}, error) {
+	prevDelta string,
+) (DeltaUpdate, map[string]string, map[string]struct{}, error) {
 	var (
 		newDeltaURL = ""
 		// TODO(ashmrtn): Eventually this should probably be a parameter so we can
 		// take in previous paths.
-		oldPaths = map[string]string{}
-		newPaths = map[string]string{}
-		excluded = map[string]struct{}{}
+		oldPaths         = map[string]string{}
+		newPaths         = map[string]string{}
+		excluded         = map[string]struct{}{}
+		invalidPrevDelta = false
+		triedPrevDelta   = false
 	)
 
 	maps.Copy(newPaths, oldPaths)
 
+	if len(prevDelta) != 0 {
+		pager.SetNext(prevDelta)
+	}
+
 	for {
 		page, err := pager.GetPage(ctx)
+
+		if !triedPrevDelta && graph.IsErrInvalidDelta(err) {
+			logger.Ctx(ctx).Infow("Invalid previous delta link", "link", prevDelta)
+
+			triedPrevDelta = true // TODO(meain): Do we need this check?
+			invalidPrevDelta = true
+
+			pager.SetNext("")
```

Comment:
hmmm I think we may actually have to adjust the pager interface for this. I think we'll need something like `Reset()` or such. I have not tested how graph SDK works, but I think if this is set to just `""` then it won't actually have a URL to send a request to. Having a `Reset` function will require either passing in the drive ID again or storing the drive ID when the pager is created. Storing the ID is probably better since it will allow the pager to be used more easily without worrying if the drive ID is also passed to the function that is trying to use it

Reply:
Good catch. I'll add a `Reset` method with `driveID` added to `driveItemPager`.